src/tests/src/study/system-model/test_component.cpp

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright 2007-2024, RTE (https://www.rte-france.com)
+ * See AUTHORS.txt
+ * SPDX-License-Identifier: MPL-2.0
+ * This file is part of Antares-Simulator,
+ * Adequacy and Performance assessment for interconnected energy networks.
+ *
+ * Antares_Simulator is free software: you can redistribute it and/or modify
+ * it under the terms of the Mozilla Public Licence 2.0 as published by
+ * the Mozilla Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Antares_Simulator is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Mozilla Public Licence 2.0 for more details.
+ *
+ * You should have received a copy of the Mozilla Public Licence 2.0
+ * along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
+ */
+
+#define WIN32_LEAN_AND_MEAN
+
+#include <boost/test/unit_test.hpp>
+
+#include "antares/study/system-model/component.h"
+
+#include "utils.h"
+
+using namespace Antares::Study::SystemModel;
+
+struct ComponentTestFixture
```

Comment:
I would rather keep the name of the test class so it is unique (to avoid compile issues if we use the same name in another cpp file of the same executable)

Reply:
Unit test have 3 steps : **Arrange**, **Act** , **Assert**.
If a piece of code belongs to **Act**, then it should be in the fixture (which is necessarily about set up / tear down)